docs/howto/how-to-make-tensorflow-plugin-with-ui/tensorflow-plugin-with-ui.md

```diff
@@ -0,0 +1,1488 @@
+        batch_size=batch_size, 
+        validation_split=0.2, 
+        subset='validation', 
+        seed=seed)
+      
+      try:
+        tf.data.Dataset.save(raw_train_ds, f"{self.data_url}/train")
+        tf.data.Dataset.save(raw_test_ds, f"{self.data_url}/test")
+        tf.data.Dataset.save(raw_val_ds, f"{self.data_url}/validation")
+        self.info("Completed data prep...")
+      except Exception as e:
+        logger.error(repr(e))
+```
+We know already `self.provider.tool_config` is simply a Python dict.
+Note, though, we have to call `int` on some values. We do this simply because we get all `tool_config` as `string` types from the UI SDK.
+So, we need to convert this to their expected types or even serialize more complex data. (that is enough for its step-by-step guide)
+
+We retrieve and parse these values and use them in `tf.keras.utils.text_dataset_from_directory(...)`. 
+This function loads our raw data into a dataset to be used by TensorFlow.
+Once we've successfully loaded them _as_ a `tf.Dataset`, we `.save(...)` them at distinct locations for our learning setup. Saving them as datasets allows us to use our data in a format optimized for TensorFlow and its training/execution.
+As previously mentioned, we recommend the [companion piece](https://www.tensorflow.org/tutorials/keras/text_classification) provided by TensorFlow for details regarding TensorFlow's API.
+There are more insights regarding TensorFlow specifics, but it's too much to cover here.
+
+Now that our data can be retrieved and loaded, we will need some way to prepare it for our model and training.
+
+### 1.3.2 `PREVIEW` mode
+
+First, we put in our new `PREVIEW` mode code and functionality.
+To better understand TensorFlow usage, please recheck the companion piece for a more detailed explanation of each line.
+_(NOTE: we will assume the reader is doing this or understands `keras` and `tensorflow` moving forward)_
+****
+
+#### Function: `send_preview_data`
+
+```python
+def send_preview_data(self):
+  raw_train_ds = tf.data.Dataset.load(f"{self.data_url}/train")
+  conf = self.provider.full_config
+  # format for table values
+  samples = []
+  text_batch, label_batch = [n for n in raw_train_ds.take(1)][0]
+  samples = {
+    'textBatch': text_batch.numpy()[:2],
+    'labelBatch': label_batch.numpy()[:2]
+  }
+  conf['Configuration']['datasetInfo']['rawTrainSample'] = samples
+
+  translation_val = conf['Configuration']['textVectorizationConfig']['translationRequest']['token']
+  if translation_val:
+    conf['Configuration']['textVectorizationConfig']['tokenChips'] = self.get_token_translation(translation_val)
+    self.info(f"Received value for translation: {str(translation_val)}")
+
+    self.provider.save_full_config(conf)
+```
+
+In brief, we load our `tf.data.Dataset` as defined previously in our `DATA` mode. Then, make some sample formats to send to our UI, or "frontend".
+Note it also calls two(2) other utility functions called...
```

Comment:
TW review:

In brief, we load the `tf.data.Dataset` as defined previously in the `DATA` mode. Then, make some sample formats to send to the UI, or "front end". Note it also calls 2 other utility functions: